### ISSUE TITLE
Underfitting problems when finetuning SAM 2 on image dataset

### ISSUE THREAD

**Issue by yujunwei04** [facebookresearch/sam2#668]
Hi all,

I've been tried to finetune SAM-2 on ground truth SA-1B & unsupervised masks (just images and point prompts) for months. No matter how I change the trianing strategies (finetune all parameters / freeze image encoder / trian a new mask token) and hyperparameters (number of masks / resolution / learning rate / optimizer), I found the training loss converge quickly within 1 or 2 epoch, which indicated a strong underfit. The evaluation results improve a little in epoch 1 but got stuck after that. I also trained to extend my training data from 3k images to 10k images, but the results are roughly the same. This seems counter-intuitive to me.

May I ask if anyone encountered the same issue, and how did you manage to overcome this? I'm also thinking about switching to SAM-v1 since it seems there are more relevant papers for finetuning & high-stars LoRA implementation available. If anyone has experience on finetuning both SAM v1 and v2, please share your experience with me! 

Thank you!!

**Comment by whistle-blower-kingdom** [comment]
Where is the high-stars LoRA implementation about SAM-v1 ?

**Comment by yujunwei04** [comment]
> Where is the high-stars LoRA implementation about SAM-v1 ?

https://github.com/JamesQFreeman/Sam_LoRA
